src/ml/neural_net/tf_activity_classifier_backend.cpp

```diff
@@ -0,0 +1,195 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+*/
+
+#include <ml/neural_net/tf_activity_classifier_backend.hpp>
+
+#include <iostream>
+#include <vector>
+
+#include <ml/neural_net/float_array.hpp>
+#include <ml/neural_net/tf_compute_context.hpp>
+#include <core/util/try_finally.hpp>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/numpy.h>
+#include <pybind11/eval.h>
+
+namespace turi {
+namespace neural_net {
+
+namespace py = pybind11;
+
+using turi::neural_net::float_array;
+using turi::neural_net::float_array_map;
+using turi::neural_net::shared_float_array;
+
+
+template <typename CallFunc> 
+auto call_pybind_function(CallFunc&& func) -> decltype(func()) {
+    PyGILState_STATE gstate;
+  	gstate = PyGILState_Ensure();
+
+    turi::scoped_finally gstate_restore([&](){PyGILState_Release(gstate);});
+
+	try {
+		func();
+	} catch(...){
+	    	log_and_throw("An error occurred!");
+	    }
+}
+
+
+static std::vector<size_t> get_dimensions( const float_array& num) {
+  std::vector<size_t> result(num.dim());
+  std::memcpy(result.data(), num.shape(), num.dim()* sizeof(size_t));
+  return result;
+}
+
+static std::vector<size_t> get_strides( const float_array& num) {
+  std::vector<size_t> result(num.dim());
+  std::fill(result.begin(), result.end(), sizeof(float));
+  return result;
+}
+
+PYBIND11_MODULE(libtctensorflow, m) {
+	py::class_<float_array>(m, "FloatArray", py::buffer_protocol())
+	.def_buffer([](float_array &m) -> py::buffer_info{
+	      return py::buffer_info(
+	          const_cast<float*>(m.data()),                   /* Pointer to buffer */
+	          sizeof(float),                                  /* Size of one scalar */
+	          py::format_descriptor<float>::format(),         /* Python struct-style format descriptor */
+	          m.dim(),                                        /* Number of dimensions */
+	          get_dimensions(m),                              /* Buffer dimensions */
+	          get_strides(m)                                  /* Strides (in bytes) for each index */
+	           
+	      );
+	});
+	py::class_<shared_float_array>(m, "SharedFloatArray", py::buffer_protocol())
```

Comment:
Is this necessary? `shared_float_array` inherits from `float_array`, at least on the C++ side

Reply:
Yes, it was not working with just `float_array`.